pkg/operator/controller/ciskubebenchreport.go

```diff
@@ -0,0 +1,300 @@
+		}
+
+		log.V(1).Info("Scheduling CIS Kubernetes Benchmark checks")
+		err = r.Client.Create(ctx, job)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("creating job: %w", err)
+		}
+
+		return ctrl.Result{}, nil
+	}
+}
+
+func (r *CISKubeBenchReportReconciler) hasReport(ctx context.Context, node *corev1.Node) (bool, error) {
+	report, err := r.ReadWriter.FindByOwner(ctx, kube.Object{Kind: kube.KindNode, Name: node.Name})
+	if err != nil {
+		return false, err
+	}
+	return report != nil, nil
+}
+
+func (r *CISKubeBenchReportReconciler) hasScanJob(ctx context.Context, node *corev1.Node) (bool, *batchv1.Job, error) {
+	jobName := r.getScanJobName(node)
+	job := &batchv1.Job{}
+	err := r.Client.Get(ctx, client.ObjectKey{Namespace: r.Config.Namespace, Name: jobName}, job)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil, nil
+		}
+		return false, nil, fmt.Errorf("getting job from cache: %w", err)
+	}
+	return true, job, nil
+}
+
+func (r *CISKubeBenchReportReconciler) newScanJob(node *corev1.Node) (*batchv1.Job, error) {
+	templateSpec, err := r.Plugin.GetScanJobSpec(*node)
+	if err != nil {
+		return nil, err
+	}
+
+	templateSpec.ServiceAccountName = r.Config.ServiceAccount
+
+	return &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.getScanJobName(node),
+			Namespace: r.Config.Namespace,
+			Labels: labels.Set{
+				kube.LabelResourceKind:        string(kube.KindNode),
+				kube.LabelResourceName:        node.Name,
+				kube.LabelK8SAppManagedBy:     kube.AppStarboardOperator,
+				kube.LabelKubeBenchReportScan: "true",
+			},
+		},
+		Spec: batchv1.JobSpec{
+			BackoffLimit:          pointer.Int32Ptr(0),
+			Completions:           pointer.Int32Ptr(1),
+			ActiveDeadlineSeconds: kube.GetActiveDeadlineSeconds(r.Config.ScanJobTimeout),
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels.Set{
+						kube.LabelResourceKind:        string(kube.KindNode),
+						kube.LabelResourceName:        node.Name,
+						kube.LabelK8SAppManagedBy:     kube.AppStarboardOperator,
+						kube.LabelKubeBenchReportScan: "true",
+					},
+				},
+				Spec: templateSpec,
+			},
+		},
+	}, nil
+}
+
+func (r *CISKubeBenchReportReconciler) getScanJobName(node *corev1.Node) string {
+	return "scan-kubebenchreports-" + node.Name
+}
+
+func (r *CISKubeBenchReportReconciler) reconcileJobs() reconcile.Func {
+	return func(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+		log := r.Logger.WithValues("job", req.NamespacedName)
+
+		job := &batchv1.Job{}
+		err := r.Client.Get(ctx, req.NamespacedName, job)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				log.V(1).Info("Ignoring job read from cache that must have been deleted")
+				return ctrl.Result{}, nil
+			}
+			return ctrl.Result{}, fmt.Errorf("getting job from cache: %w", err)
+		}
+
+		if len(job.Status.Conditions) == 0 {
+			log.V(1).Info("Job has no conditions despite using predicate")
```

Comment:
I found in tests that even though we use predicate, this condition does not always hold. Therefore, I'm double checking here. Very defensive programming, otherwise we'll have panics.